### DISCUSSION TITLE
how to compare dynamic array/slice by data pointer and length

### DISCUSSION THREAD

**Discussion by gco-bmx** [odin-lang/Odin#2786] [Q&A] [answered]
can I tell if 2 slices or 2 dynamic arrays are the same? without comparing each element
(don't want to compare each element)

As I understand, slices and dynamic arrays contain, data pointer, length, and capacity (in dynamic array case)
and behave like pointers when being assigned, i.e. the fields are copied but not the underlying data
I can see there are len() and cap() functions
is there a function to get the pointer to data pointer? so I can check if they are the same without checking each element
I haven't been able to find any answer in the documentation

an example use case is 
given an unsorted list/array of dynamic arrays, 
sort the list, 
which index did the n th element end up on?

**Comment by leidegre** [comment, marked as answer]
If you wanna know if they point to the same memory I think you want to compare the underlying pointers with `raw_data`?

```odin
package exp

import "core:fmt"
import "core:testing"

@(test)
test_raw_data :: proc(t: ^testing.T) {
	tmp: [64]u8

	x := tmp[:]
	y := tmp[:]
	z := tmp[32:]

	fmt.println(x, y, z)

	a := raw_data(x)
	b := raw_data(y)
	c := raw_data(z)

	fmt.println(a, b, c)
}
```

```
> odin test .\exp\raw_data.odin -file
[Package: exp]
[Test: test_raw_data]
[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
0x9288AFF400 0x9288AFF400 0x9288AFF420
[test_raw_data : SUCCESS]
----------------------------------------
1/1 SUCCESSFUL
```

**Reply by gco-bmx** [reply]
Argh Thanks
